pkg/controller/release.go

```diff
@@ -240,8 +249,26 @@ func (e *helmExternal) Update(ctx context.Context, mg resource.Managed) (managed
 		return managed.ExternalUpdate{}, errors.New(errNotRelease)
 	}
 
+	if cr.Spec.RollbackLimit != nil &&
+		cr.Status.Synced && cr.Status.AtProvider.State == release.StatusFailed {
+		e.logger.Debug("Last release failed")
+		if cr.Status.Failed < *cr.Spec.RollbackLimit {
+			// Rollback
+			e.logger.Debug("Will rollback to retry")
+			cr.Status.Failed++
+			if cr.Status.AtProvider.Revision < 2 {
```

Comment:
If revision is 1 (or less than 2 in the original impl :) ), rollback wouldn't work since there is no previous revision to rollback to. So, we need to check and uninstall in this case and use rollback otherwise.